starfish/core/_display.py

```diff
@@ -190,49 +194,52 @@ def display(
     Notes
     -----
     - To use in ipython, use the `%gui qt5` magic.
-    - Napari axes currently cannot be labeled. Until such a time that they can, this function will
+    - napari axes currently cannot be labeled. Until such a time that they can, this function will
       order them by Round, Channel, and Z.
-    - Requires napari 0.0.6: install starfish using `pip install starfish[napari]` to install all
+    - Requires napari 0.0.9: install starfish using `pip install starfish[napari]` to install all
       necessary requirements
     """
     if stack is None and spots is None:
+        # masks without stack or spots have no context so don't check for that
         raise TypeError("expected a stack and/or spots; got nothing")
 
-    if Window is None or Viewer is None:
-        warnings.warn("Requires napari 0.0.6. Run `pip install starfish[napari]` to install the "
-                      "necessary requirements.")
-        return
-
-    from PyQt5.QtWidgets import QApplication
+    try:
+        import napari
+        if napari.__version__ != NAPARI_VERSION:
+            raise Exception
+    except Exception:
```

Comment:
This is overly broad.  If there are other issues (perhaps `napari.__version__` doesn't exist?), it is improperly caught.

Reply:
This is intentional since we pinned napari to a specific version. `napari.__version__` not existing or some other unforeseen error occurring would mean that it is not behaving as expected and must not be the pinned version. Do you still think this is bad?

Reply:
I always prefer to catch the most specific exception possible because I don't want to mask some other error that may warrant a different fix.  In this case, your point makes a lot of sense for end users.

Two places this breaks down:
1. A developer who is experimenting with the starfish and napari bridge may find this problematic however.  Imagine that I have napari installed from a working tree, and I'm mucking around with the code there.  I could introduce a python parse error that this completely masks.
2. Someone introduces more code into this try-block, and doesn't think through the ramifications of having a catch-all exception handler around it.

tl;dr, I'm ok with you leaving this as is, because it's not a terribly common scenario that someone would encounter.  Be aware of doing general catch-all exception handlers though.

Reply:
What do you think about a catch-all that prepends the raised error message prior to the spiel about an incorrect napari version?

Reply:
If I were building this, I would do, (in pseudocode):

```
try:
    import napari
except ImportError:
    raise ImportError(f"Requires napari {NAPARI_VERSION...")

try:
    version = napari.__version__
except Exception as ex:
    raise Exception("Could not identify napari version") from ex

if version != NAPARI_VERSION:
        raise Exception(f"Incorrect version of napari installed ({version}).  Please run ...")
```

I think your proposed solution works too, or you could just chain the exception:

```
except Exception as ex:
    raise ImportError(f"...") from ex
```